Fails to build on x86_64-linux

<details><summary>End of build log</summary>


```console
ente-auth> [  +63 ms] Building Linux application...
ente-auth> [   +7 ms] executing: [build/linux/x64/release/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DFLUTTER_TARGET_PLATFORM=linux-x64 /build/source/auth/linux
ente-auth> [ +739 ms] -- The CXX compiler identification is Clang 19.1.7
ente-auth> [  +34 ms] -- Detecting CXX compiler ABI info
ente-auth> [ +323 ms] -- Detecting CXX compiler ABI info - done
ente-auth> [  +93 ms] -- Check for working CXX compiler: /nix/store/32jfd5s845ys74nkzlgsh7cnq83y4lri-clang-wrapper-19.1.7/bin/clang++ - skipped
ente-auth> [   +9 ms] -- Detecting CXX compile features
ente-auth> [   +2 ms] -- Detecting CXX compile features - done
ente-auth> [ +132 ms] -- Found PkgConfig: /nix/store/bbp56ai8w43xkfisar83h37imjdlcx6r-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
ente-auth> [        ] -- Checking for module 'gtk+-3.0'
ente-auth> [  +62 ms] --   Found gtk+-3.0, version 3.24.49
ente-auth> [ +504 ms] -- Checking for module 'glib-2.0'
ente-auth> [  +34 ms] --   Found glib-2.0, version 2.82.5
ente-auth> [  +83 ms] -- Checking for module 'gio-2.0'
ente-auth> [  +34 ms] --   Found gio-2.0, version 2.82.5
ente-auth> [ +145 ms] -- Checking for module 'libsecret-1>=0.18.4'
ente-auth> [  +39 ms] --   Found libsecret-1, version 0.21.7
ente-auth> [ +181 ms] -- Checking for module 'ayatana-appindicator3-0.1'
ente-auth> [  +50 ms] --   Found ayatana-appindicator3-0.1, version 0.5.92
ente-auth> [ +692 ms] -- The C compiler identification is Clang 19.1.7
ente-auth> [  +24 ms] -- Detecting C compiler ABI info
ente-auth> [ +267 ms] -- Detecting C compiler ABI info - done
ente-auth> [  +97 ms] -- Check for working C compiler: /nix/store/32jfd5s845ys74nkzlgsh7cnq83y4lri-clang-wrapper-19.1.7/bin/clang - skipped
ente-auth> [        ] -- Detecting C compile features
ente-auth> [   +1 ms] -- Detecting C compile features - done
ente-auth> [  +89 ms] CMake Error at /nix/store/ad4c8nx2lkjs4zq0sj9jdpn7w45v9ccs-cmake-3.31.6/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
ente-auth> [   +1 ms]   Could NOT find Java (missing: Java_JAVA_EXECUTABLE Java_JAR_EXECUTABLE
ente-auth> [        ]   Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE)
ente-auth> [        ] Call Stack (most recent call first):
ente-auth> [        ]   /nix/store/ad4c8nx2lkjs4zq0sj9jdpn7w45v9ccs-cmake-3.31.6/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
ente-auth> [        ]   /nix/store/ad4c8nx2lkjs4zq0sj9jdpn7w45v9ccs-cmake-3.31.6/share/cmake-3.31/Modules/FindJava.cmake:326 (find_package_handle_standard_args)
ente-auth> [        ]   flutter/ephemeral/.plugin_symlinks/jni/src/CMakeLists.txt:31 (find_package)
ente-auth> [        ] -- Configuring incomplete, errors occurred!
ente-auth> [  +18 ms] Building Linux application... (completed in 3.7s)
ente-auth> [   +1 ms] "flutter linux" took 5,395ms.
ente-auth> [   +3 ms] Unable to generate build files
ente-auth> [        ]
ente-auth>            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
ente-auth>            #1      _runCmake (package:flutter_tools/src/linux/build_linux.dart:174:5)
ente-auth>            <asynchronous suspension>
ente-auth>            #2      buildLinux (package:flutter_tools/src/linux/build_linux.dart:81:5)
ente-auth>            <asynchronous suspension>
ente-auth>            #3      BuildLinuxCommand.runCommand (package:flutter_tools/src/commands/build_linux.dart:87:5)
ente-auth>            <asynchronous suspension>
ente-auth>            #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1408:27)
ente-auth>            <asynchronous suspension>
ente-auth>            #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
ente-auth>            <asynchronous suspension>
ente-auth>            #6      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
ente-auth>            <asynchronous suspension>
ente-auth>            #7      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:416:9)
ente-auth>            <asynchronous suspension>
ente-auth>            #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
ente-auth>            <asynchronous suspension>
ente-auth>            #9      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:364:5)
ente-auth>            <asynchronous suspension>
ente-auth>            #10     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:130:9)
ente-auth>            <asynchronous suspension>
ente-auth>            #11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
ente-auth>            <asynchronous suspension>
ente-auth>            #12     main (package:flutter_tools/executable.dart:93:3)
ente-auth>            <asynchronous suspension>
ente-auth>
ente-auth>
ente-auth> [   +1 ms] Running 1 shutdown hook
ente-auth> [        ] Shutdown hooks complete
ente-auth> [   +3 ms] exiting with code 1
```
</details>

--- a/pkgs/by-name/en/ente-auth/package.nix
+++ b/pkgs/by-name/en/ente-auth/package.nix
@@ -8,6 +8,7 @@
   makeDesktopItem,
   copyDesktopItems,
   makeWrapper,
+  jdk17_headless,
 }:
 let
   # fetch simple-icons directly to avoid cloning with submodules,
@@ -17,14 +18,14 @@ let
 in
 flutter324.buildFlutterApplication rec {
   pname = "ente-auth";
-  version = "4.3.2";
+  version = "4.3.5";
 
   src = fetchFromGitHub {
     owner = "ente-io";
     repo = "ente";
     sparseCheckout = [ "auth" ];
     tag = "auth-v${version}";
-    hash = "sha256-/WWodQcMibwXVexI+XbTZYRkIMtfNHk3bJVBPJHcoqI=";
+    hash = "sha256-kM1y3Q5Z8J84qHhki9A+I/uY7xYQNMlfh2ZhxzpUBHM=";
   };
 
   sourceRoot = "${src.name}/auth";
@@ -51,6 +52,12 @@ flutter324.buildFlutterApplication rec {
     webkitgtk_4_0
     sqlite
     libayatana-appindicator
+    # The networking client used by ente-auth (native_dio_adapter)
+    # introduces a transitive dependency on Java, which technically
+    # is only needed for the Android implementation.
+    # Unfortunately, attempts to remove it from the build entirely were
+    # unsuccessful.
+    jdk17_headless # JDK version used by upstream CI
   ];
 
   # Based on https://github.com/ente-io/ente/blob/main/auth/linux/packaging/rpm/make_config.yaml
@@ -84,6 +91,9 @@ flutter324.buildFlutterApplication rec {
 
     # For backwards compatibility
     ln -s $out/bin/enteauth $out/bin/ente_auth
+
+    # Not required at runtime as it's only used on Android
+    rm $out/app/ente-auth/lib/libdartjni.so
   '';
 
   passthru.updateScript = ./update.sh;

--- a/pkgs/by-name/en/ente-auth/pubspec.lock.json
+++ b/pkgs/by-name/en/ente-auth/pubspec.lock.json
@@ -356,6 +356,16 @@
       "source": "hosted",
       "version": "3.1.1"
     },
+    "cronet_http": {
+      "dependency": "transitive",
+      "description": {
+        "name": "cronet_http",
+        "sha256": "3af9c4d57bf07ef4b307e77b22be4ad61bea19ee6ff65e62184863f3a09f1415",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "1.3.2"
+    },
     "cross_file": {
       "dependency": "transitive",
       "description": {
@@ -385,6 +395,16 @@
       },
       "source": "hosted",
       "version": "1.0.0"
+    },
+    "cupertino_http": {
+      "dependency": "transitive",
+      "description": {
+        "name": "cupertino_http",
+        "sha256": "6fcf79586ad872ddcd6004d55c8c2aab3cdf0337436e8f99837b1b6c30665d0c",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "2.0.2"
     },
     "dart_style": {
       "dependency": "transitive",
@@ -430,11 +450,11 @@
       "dependency": "direct main",
       "description": {
         "name": "dio",
-        "sha256": "5598aa796bbf4699afd5c67c0f5f6e2ed542afc956884b9cd58c306966efc260",
+        "sha256": "253a18bbd4851fecba42f7343a1df3a9a4c1d31a2c1b37e221086b4fa8c8dbc9",
         "url": "https://pub.dev"
       },
       "source": "hosted",
-      "version": "5.7.0"
+      "version": "5.8.0+1"
     },
     "dio_web_adapter": {
       "dependency": "transitive",
@@ -1073,6 +1093,16 @@
       "source": "hosted",
       "version": "4.0.2"
     },
+    "http_profile": {
+      "dependency": "transitive",
+      "description": {
+        "name": "http_profile",
+        "sha256": "7e679e355b09aaee2ab5010915c932cce3f2d1c11c3b2dc177891687014ffa78",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "0.1.0"
+    },
     "image": {
       "dependency": "transitive",
       "description": {
@@ -1102,6 +1132,16 @@
       },
       "source": "hosted",
       "version": "1.0.4"
+    },
+    "jni": {
+      "dependency": "transitive",
+      "description": {
+        "name": "jni",
+        "sha256": "f377c585ea9c08d48b427dc2e03780af2889d1bb094440da853c6883c1acba4b",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "0.10.1"
     },
     "js": {
       "dependency": "transitive",
@@ -1323,6 +1363,16 @@
       "source": "hosted",
       "version": "1.0.2"
     },
+    "native_dio_adapter": {
+      "dependency": "direct main",
+      "description": {
+        "name": "native_dio_adapter",
+        "sha256": "7420bc9517b2abe09810199a19924617b45690a44ecfb0616ac9babc11875c03",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "1.4.0"
+    },
     "nested": {
       "dependency": "transitive",
       "description": {
@@ -1342,6 +1392,16 @@
       },
       "source": "hosted",
       "version": "0.5.0"
+    },
+    "objective_c": {
+      "dependency": "transitive",
+      "description": {
+        "name": "objective_c",
+        "sha256": "62e79ab8c3ed6f6a340ea50dd48d65898f5d70425d404f0d99411f6e56e04584",
+        "url": "https://pub.dev"
+      },
+      "source": "hosted",
+      "version": "4.1.0"
     },
     "otp": {
       "dependency": "direct main",

